### ISSUE TITLE
no-string-refs: template strings are ignored

### ISSUE THREAD

**Issue by MayhemYDG** [yannickcr/eslint-plugin-react#2148]
```jsx
<Button ref={`ref-${id}`} />
```

It seems like the `no-string-refs` rule doesn't catch template strings, is this a limitation of the parser/static analysis, or can this be fixed perhaps?

**Comment by ljharb** [comment]
It definitely should!

**Comment by jenil94** [comment]
@ljharb I can work on this issue if it is not assigned to anyone

**Comment by ljharb** [comment]
@jenil94 that'd be great, thanks
